scripts/validate_docs_snippet_line_numbers.py

```diff
@@ -0,0 +1,200 @@
+import glob
+import logging
+import os
+import re
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, List, Tuple
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+OPENING_TAG: str = "<snippet>"
+CLOSING_TAG: str = "</snippet>"
+DOCUSAURUS_REGEX: re.Pattern = re.compile(r"```python file=(.+)#L(\d*)(?:-L)?(\d*)")
+
+
+@dataclass
+class DocusaurusRef:
+    docs_file: str
+    py_file: str
+    line_numbers: Tuple[int, int]
+
+    @classmethod
+    def from_regex_parsed_values(
+        cls, file: str, values: Tuple[str, str, str]
+    ) -> "DocusaurusRef":
+        path, start, end = values
+
+        start_line: int = int(start)
+        end_line: int = int(end) if end else start_line
+        line_numbers: Tuple[int, int] = start_line, end_line
```

Comment:
FYI This line was failing for me, maybe because of python 3.7 I needed to add parens `(start_line, end_line)`. I think it's ok to leave as-is, just FYI.